graph/db/sql_store.go

```diff
@@ -1549,3 +1669,374 @@ func upsertChanPolicyExtraSignedFields(ctx context.Context, db SQLQueries,
+		),
+		MessageFlags:  msgFlags,
+		ChannelFlags:  chanFlags,
+		TimeLockDelta: uint16(dbPolicy.Timelock),
+		MinHTLC: lnwire.MilliSatoshi(
+			dbPolicy.MinHtlcMsat,
+		),
+		MaxHTLC: lnwire.MilliSatoshi(
+			dbPolicy.MaxHtlcMsat.Int64,
+		),
+		FeeBaseMSat: lnwire.MilliSatoshi(
+			dbPolicy.BaseFeeMsat,
+		),
+		FeeProportionalMillionths: lnwire.MilliSatoshi(dbPolicy.FeePpm),
+		ToNode:                    toNode,
+		InboundFee:                inboundFee,
+		ExtraOpaqueData:           recs,
+	}, nil
+}
+
+// extractChannelPolicies extracts the sqlc.ChannelPolicy records from the give
+// row which is expected to be a sqlc type that contains channel policy
+// information. It returns two policies, which may be nil if the policy
+// information is not present in the row.
+//
+//nolint:ll
+func extractChannelPolicies(row any) (*sqlc.ChannelPolicy, *sqlc.ChannelPolicy,
+	error) {
+
+	var policy1, policy2 *sqlc.ChannelPolicy
+	switch r := row.(type) {
+	case sqlc.ListChannelsByNodeIDRow:
+		if r.Policy1ID.Valid {
+			policy1 = &sqlc.ChannelPolicy{
+				ID:                      r.Policy1ID.Int64,
+				Version:                 r.Policy1Version.Int16,
+				ChannelID:               r.ID,
+				NodeID:                  r.Policy1NodeID.Int64,
+				Timelock:                r.Policy1Timelock.Int32,
+				FeePpm:                  r.Policy1FeePpm.Int64,
+				BaseFeeMsat:             r.Policy1BaseFeeMsat.Int64,
+				MinHtlcMsat:             r.Policy1MinHtlcMsat.Int64,
+				MaxHtlcMsat:             r.Policy1MaxHtlcMsat,
+				LastUpdate:              r.Policy1LastUpdate,
+				InboundBaseFeeMsat:      r.Policy1InboundBaseFeeMsat,
+				InboundFeeRateMilliMsat: r.Policy1InboundFeeRateMilliMsat,
+				Disabled:                r.Policy1Disabled,
+				Signature:               r.Policy1Signature,
+			}
+		}
+		if r.Policy2ID.Valid {
+			policy2 = &sqlc.ChannelPolicy{
+				ID:                      r.Policy2ID.Int64,
+				Version:                 r.Policy2Version.Int16,
+				ChannelID:               r.ID,
+				NodeID:                  r.Policy2NodeID.Int64,
+				Timelock:                r.Policy2Timelock.Int32,
+				FeePpm:                  r.Policy2FeePpm.Int64,
+				BaseFeeMsat:             r.Policy2BaseFeeMsat.Int64,
+				MinHtlcMsat:             r.Policy2MinHtlcMsat.Int64,
+				MaxHtlcMsat:             r.Policy2MaxHtlcMsat,
+				LastUpdate:              r.Policy2LastUpdate,
+				InboundBaseFeeMsat:      r.Policy2InboundBaseFeeMsat,
+				InboundFeeRateMilliMsat: r.Policy2InboundFeeRateMilliMsat,
+				Disabled:                r.Policy2Disabled,
+				Signature:               r.Policy2Signature,
+			}
+		}
+
+		return policy1, policy2, nil
+	default:
+		return nil, nil, fmt.Errorf("unexpected row type in "+
+			"extractChannelPolicies: %T", r)
+	}
+}
+
+// extractChannel extracts the sqlc.Channel record from the given row
+// which is expected to be a sqlc type that contains channel information.
+func extractChannel(row any) (sqlc.Channel, error) {
+	switch r := row.(type) {
```

Comment:
I think most of these type switches and explicit struct copies can be eliminated by using the embed feature, see my other comment.